src/metabase/server/middleware/session.clj

```diff
@@ -305,3 +307,89 @@
   `(do-with-current-user
     (with-current-user-fetch-user-for-id ~current-user-id)
     (fn [] ~@body)))
+
+
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                              check-session-timeout                                             |
+;;; +----------------------------------------------------------------------------------------------------------------+
+
+(defsetting session-timeout
+  ;; Should be in the form {:amount 60 :unit "minutes"} where the unit is one of "seconds", "minutes" or "hours". The amount is nillable.
+  (deferred-tru "Time before inactive users are logged out. By default, sessions last indefinitely.")
+  :type       :json
+  :default    nil)
+
+(defn session-timeout->seconds
+  "Convert a session timeout setting to seconds."
+  [{:keys [unit amount]}]
+  (when amount
+    (-> (case unit
+          "seconds" amount
+          "minutes" (* amount 60)
+          "hours"  (* amount 3600))
+        (max 60)))) ; Ensure a minimum of 60 seconds so a user can't lock themselves out
+
+(defn response-with-session-timeout-cookie
+  "Adds a cookie to the response that expires after `session-timeout-seconds` seconds."
+  [request request-time timeout-seconds response]
+  (cond
+    ;; If there is no session cookie because the user is logged out, clear the timeout cookie
+    (and (get-in request [:cookies metabase-session-timeout-cookie :value])
```

Comment:
It should set the cookie only if the BE session is alive and when it gets removed it should remove the cookie

Reply:
> when it gets removed it
Can you be more specific on `it` here? Do you mean when the session cookie is removed from logging out? I intend to refactor this so the session cookie gets removed in the same function as the timeout cookie on logout.

Reply:
I mean the back end should set cookie after verifying that a BE session exists and only in this case. It should not check the existence of other cookies. Maybe I misunderstood the code here because I don't know Clojure at all